src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

```diff
@@ -1210,12 +1184,13 @@ public ValueTask<int> SendToAsync(Socket socket, CancellationToken cancellationT
                     ValueTask.FromException<int>(CreateException(error));
             }
 
-            public ValueTask ConnectAsync(Socket socket, bool saeaCancelable)
+            public ValueTask ConnectAsync(Socket socket, CancellationToken cancellationToken)
             {
                 try
                 {
-                    if (socket.ConnectAsync(this, userSocket: true, saeaCancelable: saeaCancelable))
+                    if (socket.ConnectAsync(this, userSocket: true, saeaCancelable: false, cancellationToken))
```

Comment:
do we need the `saeaCancelable` since we are forwarding the token?

Reply:
`saeaCancelable: true` means we are using a non-Task-based methods. It is forwarded to `StartOperationConnect` as `saeaMultiConnectCancelable` and it ensures setting `_multipleConnectCancellation` which is needed to support `CancelConnectAsync`:

https://github.com/dotnet/runtime/blob/52a447aac269964ea45ca20fd100b4888555622e/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs#L602-L607

Later, there is an assertion that guarantees that the two are mutually exclusive:

https://github.com/dotnet/runtime/blob/52a447aac269964ea45ca20fd100b4888555622e/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs#L687-L692

I'm happy to rename the variable if `saeaCancelable` is confusing.

Reply:
maybe rename or adding comment with explanation would be good.